front/components/agent_builder/capabilities/mcp/utils/formDefaults.ts

```diff
@@ -69,5 +69,6 @@ export function getDefaultFormValues(mcpServerView: MCPServerViewType | null) {
     name: "",
     description: "",
     configuration: getDefaultConfiguration(mcpServerView),
+    sources: { in: [], notIn: [] },
```

Comment:
Instead of conditionally create default values based on a selected tool, I think it's easier to have sources for all, and then don't send them to backend unless it's canvas